### PR TITLE
Adding support for `$TSFixMeAny`

### PR DESCRIPTION
**Pull request by ghost**
### Summary

In this PR, I add support for the `$TSFixMeAny` type that we have in `v2/core/util/flowCompat`. The common change in this PR is replacing `t.tsAnyKeyword()` and with `FlowCompatTypes.any`.

I have also removed the portion of the script that converted `JestMockFn` to `jest.MockedFunction` because I was having trouble getting the tests to pass after my code changes. No need to worry though, https://github.com/FB-PLP/traject/pull/11185 removes all instances of `JestMockFn`

[sc-199791]

### Test Plan:
- [x]  yarn commands still work and all tests pass. For example `yarn build`, `yarn test`, and `yarn types`

### CODE REVIEW

File: src/convert/utils/type-mappings.ts [hunk 1]
```diff
@@ -30,3 +32,7 @@ export const SyntheticEvents = {
 export const MomentTypes = {
   MomentDuration: "Duration",
 } as const;
+
+export const FlowCompatTypes = {
+  any: t.tsTypeReference(t.identifier("$TSFixMeAny")),
+};
```
Comment (jeremiah-clothier) [inline src/convert/utils/type-mappings.ts:38]: For reference, this is how I created a `$TSFixMeAny` type declaration

File: src/convert/utils/configurable-type-provider.ts [hunk 2]
```diff
@@ -21,18 +22,21 @@ export class ConfigurableTypeProvider {
...
       return t.tsTypeReference(
         t.identifier("Record"),
-        t.tsTypeParameterInstantiation([t.tsAnyKeyword(), t.tsAnyKeyword()])
+        t.tsTypeParameterInstantiation([
+          FlowCompatTypes.any,
+          FlowCompatTypes.any,
+        ])
       );
     } else {
-      return t.tsAnyKeyword();
```
Comment (jeremiah-clothier) [inline src/convert/utils/configurable-type-provider.ts:27]: Basically, looked for `t.tsAnyKeyword()` and replaced if with `FlowCompatTypes.any`

File: src/convert/declarations.test.ts [hunk 3]
```diff
@@ -225,21 +228,22 @@ describe("transform declarations", () => {
     class Impl extends Base<Function, {}> {}
     `;
     const expected = dedent`
-    import type {$TSFixMeFunction} from 'v2/core/util/flowCompat';
-    class Base<P = any, S = any> {}
-    class Impl extends Base<$TSFixMeFunction, Record<any, any>> {}
+    import type {$TSFixMeFunction, $TSFixMeAny} from 'v2/core/util/flowCompat';
```
Comment (ghost) [inline src/convert/declarations.test.ts:231]: Nice to have these tests and confirm that it handles multiple type imports correctly

File: src/test/regression/regression.test.ts [hunk 4]
```diff
@@ -97,6 +97,10 @@ function replaceImports(data: string) {
       `'${require.resolve(
         "../../../node_modules/@types/react/index.d"
       )}'`.replace(/.d.ts/, "")
+    )
+    .replace(
+      /v2\/core\/util\/flowCompat/,
+      require.resolve("../../../src/flowCompat.ts").replace(/.ts/, "")
```
Comment (ghost) [inline src/test/regression/regression.test.ts:103]: > // The in memory typescript compiler can't seem to find imported modules.
// This hack to replace imports with absolute paths seems to work.
// Don't apply this before the snapshot, or your snapshot will have absoulte paths

source: existing code comment in lines above this diff

**Review by ahuth — APPROVED** [review]
LGTM 🤘

**Pull request merged**